e2e/testcases/otel_collector_test.go

```diff
@@ -0,0 +1,260 @@
+		ksa := fake.ServiceAccountObject(DefaultMonitorKSA)
+		if err := nt.KubeClient.Get(DefaultMonitorKSA, ocmetrics.MonitoringNamespace, ksa); err != nil {
+			nt.T.Log(fmt.Sprintf("service account not found during cleanup: %v", err))
+		}
+		_, err := unsetAnnotation(nt.KubeClient, "iam.gke.io/gcp-service-account", ksa)
+		if err != nil {
+			nt.T.Log(err)
+		}
+	})
+
+	// If Workload Identity enabled on cluster, setup KSA to GSA annotation
+	if workloadPool, err := getWorkloadPool(nt); err != nil {
+		nt.T.Fatal(err)
+	} else if workloadPool != "" {
+		nt.T.Log(fmt.Sprintf("Workload identity enabled, adding KSA annotation to use %s service account", MonitorGSA))
+		ksa := fake.ServiceAccountObject(DefaultMonitorKSA)
+		if err := nt.KubeClient.Get(DefaultMonitorKSA, ocmetrics.MonitoringNamespace, ksa); err != nil {
+			nt.T.Fatal(fmt.Sprintf("service account does not exist: %v", err))
+		}
+		if _, err := setAnnotation(nt.KubeClient, "iam.gke.io/gcp-service-account",
+			fmt.Sprintf("%s@%s.iam.gserviceaccount.com", MonitorGSA, *e2e.GCPProject), ksa); err != nil {
+			nt.T.Fatal(err)
+		}
+	}
+
+	nt.T.Log("Restart otel-collector pod to refresh the ConfigMap, log and IAM")
+	nomostest.DeletePodByLabel(nt, "app", ocmetrics.OpenTelemetry, false)
+	if err := nt.Watcher.WatchForCurrentStatus(kinds.Deployment(), ocmetrics.OtelCollectorName, ocmetrics.MonitoringNamespace); err != nil {
+		nt.T.Fatal(err)
+	}
+
+	nt.T.Log("Watch for metrics in GCM, timeout 2 minutes")
+	ctx := context.Background()
+	client, err := createGCMClient(ctx)
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+	startTime := time.Now().UTC()
+	// retry for 2 minutes until metric is accessible from GCM
+	_, err = retry.Retry(120*time.Second, func() error {
+		for _, metricType := range GCMMetricTypes {
+			descriptor := fmt.Sprintf("%s/%s", GCMMetricPrefix, metricType)
+			it := listMetricInGCM(ctx, nt, client, startTime, descriptor)
+			return validateMetricInGCM(nt, it, descriptor, nt.ClusterName)
+		}
+		return nil
+	})
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+
+	nt.T.Log("Checking the otel-collector log contains no failure...")
+	err = validateDeploymentLogHasNoFailure(nt, ocmetrics.OtelCollectorName, ocmetrics.MonitoringNamespace, GCMExportErrorCaption)
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+
+	// The ConfigMap that is expected to trigger duplicate time series error has
+	// name 'otel-collector-custom', which by the setup in otel-collector deployment
+	// will take precedence over 'otel-collector-googlecloud' that was deployed
+	// by the test.
+	nt.T.Log("Apply custom otel-collector ConfigMap that could cause duplicate time series error")
+	nt.MustKubectl("apply", "-f", "../testdata/otel-collector/otel-cm-duplicate-timeseries.yaml")
+	nt.T.Log("Restart otel-collector pod to refresh the ConfigMap and log")
+	nomostest.DeletePodByLabel(nt, "app", ocmetrics.OpenTelemetry, false)
+	if err := nt.WatchForAllSyncs(); err != nil {
```

Comment:
nit: I think this `WatchForAllSyncs` is most likely a no-op and can be removed, since `WatchForAllSyncs` won't wait for the otel Pod to restart. The `WatchForCurrentStatus` below should be sufficient AFAICT.

Reply:
Removed!